### PR TITLE
Performance metrics computation methodology

### LINKED ISSUE #3864: No metrics methodology document
#### Problem

The way TPS and confirmation times are measured sometimes changes. 

#### Proposed Solution

Add a document to git that describes precisely how the metrics are calculated. Reference that document in the "TPS Report" section of the release notes.

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
No metrics methodology document

#### Summary of Changes
Added the document.

Fixes #3864

### CODE REVIEW

File: book/src/performance-metrics.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Each cluster node maintains various counters that are incremented on certain events.
+These counters are periodically uploaded to a cloud based database. Solana's metrics
+dashboard fetches these counters, and computes the performance metrics and displays
+it on the dashboard. 
+
+## TPS
+
+The leader node's bank maintains a count of transactions that it processed. The dashboard
+displays sum of the count for every 2 second period in the TPS time series graph.
```
Comment (aeyakovenko) [inline book/src/performance-metrics.md]: 2 second sum?

File: book/src/performance-metrics.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+The dashboard also shows mean, maximum and total TPS as a running counter.
```
Comment (aeyakovenko) [inline book/src/performance-metrics.md]: Is the mean a 2 second or 1 second mean?

File: book/src/performance-metrics.md (hunk 1 above)
Comment (pgarg66) [reply book/src/performance-metrics.md]: I should have clarified more. Sum over 2 seconds, and then sum is divided by 2. It's basically how the current dashboard does it.

File: book/src/performance-metrics.md (hunk 2 above)
Comment (pgarg66) [reply book/src/performance-metrics.md]: It's 1 second mean

File: book/src/performance-metrics.md (hunk 1 above)
Comment (pgarg66) [reply book/src/performance-metrics.md]: ```SELECT sum("count") / 2 AS "errors" FROM "$testnet"."autogen"."counter-bank-process_transactions-error_count" WHERE $timeFilter GROUP BY time(2s) FILL(0)```

File: book/src/performance-metrics.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Confimation Time
+
+Each validator node maintains a list of active ledger forks that are visible
+to the node. Whenever a fork's final tick is registered, the fork is frozen, and
```
Comment (aeyakovenko) [inline book/src/performance-metrics.md]: That’s not correct.  The validator confirmation time for a fork is based on the difference when a fork is created and when the first child fork that has a cumulative supermajority vote for the original fork s frozen.

Comment (pgarg66) [reply book/src/performance-metrics.md]: I understand that fork is frozen when there's a cumulative supermajority vote. I don't understand why the child fork has to be frozen?

Comment (aeyakovenko) [reply book/src/performance-metrics.md]: Frozen right now means that blocktree received all the blobs and the bank processed them.  Validator confirmation time for a parent is based on when any descendant of the parent is frozen and that branch has a supermajority conforming the parent.  Replay stage doesn’t compute a rolling vote total on incomplete forks.

Comment (pgarg66) [reply book/src/performance-metrics.md]: ok. that makes sense. I had interpreted your last comment as one of the child fork has to receive supermajority vote. Just to confirm, this is when a fork is considered confirmed:

1. The fork itself has received cumulative supermajority vote.
2. One of the child of the fork has been frozen (i.e. received all it's blocks)

File: book/src/performance-metrics.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Performance Metrics
+
+Solana cluster performance is measured as average number of transactions per second
+that the network can sustain (TPS). And, how long it takes for a transcation to be
+confirmed by super majoring of the cluster (Confirmation Time).
```
Comment (mvines) [inline book/src/performance-metrics.md]: "super majoring" -> "a super majority"

Comment (pgarg66) [reply book/src/performance-metrics.md]: oops

**Pull request merged**